packages/@vue/cli-service/lib/util/loadFileConfig.js

```diff
@@ -1,34 +1,30 @@
-const fs = require('fs')
-const path = require('path')
+// @ts-check
+const { existsSync } = require('fs')
+const { join } = require('path')
+const { register } = require('esbuild-register/dist/node')
 
-const isFileEsm = require('is-file-esm')
-const { loadModule } = require('@vue/cli-shared-utils')
+const POSSIBLE_CONFIG_PATHS = [
+  process.env.VUE_CLI_SERVICE_CONFIG_PATH,
+  'vue.config.js',
+  'vue.config.cjs',
+  'vue.config.mjs',
+  'vue.config.ts'
+].filter(i => !!i)
 
 module.exports = function loadFileConfig (context) {
-  let fileConfig, fileConfigPath
+  let fileConfig
 
-  const possibleConfigPaths = [
-    process.env.VUE_CLI_SERVICE_CONFIG_PATH,
-    './vue.config.js',
-    './vue.config.cjs',
-    './vue.config.mjs'
-  ]
-  for (const p of possibleConfigPaths) {
-    const resolvedPath = p && path.resolve(context, p)
-    if (resolvedPath && fs.existsSync(resolvedPath)) {
-      fileConfigPath = resolvedPath
-      break
-    }
-  }
+  const formatPath = p => join(context, p)
+  const fileConfigPath = POSSIBLE_CONFIG_PATHS.map(formatPath).find(p => existsSync(p))
+  const compatESModuleRequire = m => m.__esModule ? m.default : m
 
   if (fileConfigPath) {
-    const { esm } = isFileEsm.sync(fileConfigPath)
-
-    if (esm) {
-      fileConfig = import(fileConfigPath)
-    } else {
-      fileConfig = loadModule(fileConfigPath, context)
-    }
+    // Use esbuild to compile conifg files
+    register({
+      target: 'es2017',
+      format: 'cjs'
```

Comment:
We need to support Node native ES modules, too. They shouldn't be transpiled to cjs.
You can take the related code in Vite as a reference: https://github.com/vitejs/vite/blob/4abade62f101049b941fa8f6b98187c5587e238f/packages/vite/src/node/config.ts#L826-L847

Reply:
got it